snf-cyclades-app/synnefo/logic/servers.py

```diff
@@ -218,6 +224,7 @@ def create_server(vm, nics, volumes, flavor, image, personality, password):
         'img_format': str(image['format']),
         'img_personality': json.dumps(personality),
         'img_properties': json.dumps(image['metadata']),
+        'auth_keys': public_key,
```

Comment:
Based on the latest version that I pushed, the pipeline of the request:

1. User makes a request at /servers
2. The `key_name` parameter is optional that means it will be pushed as `None` to the next layers if omitted.
3. The `key_name` parameter is pushed onto the logic layer
4. If it is `None` then the `public_key` parameter will also be `None`. If not the system will try to retrieve the public key from the database. If the tuple (`user_id`, `key_name`) is not found, a 404 error will be raised. Else `public_key` is set to the public key of the user and then pushed to the creation layer
5. If `public_key` is None then `auth_keys` will be omitted. Else it will be inserted to the parameters of Image